src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java

```diff
@@ -0,0 +1,145 @@
+   *     that EST is 240 minutes behind UTC.
+   * @throws IllegalArgumentException if the magnitude of timezoneOffsetMinutes is greater than 720.
+   */
+  // TODO: Create a timeSlotsForWeek method.
+  public static List<AvailabilityTimeSlot> timeSlotsForDay(
+      Instant instant, int timezoneOffsetMinutes) {
+    Preconditions.checkArgument(
+        Math.abs(timezoneOffsetMinutes) <= 720,
+        "Offset greater than 720 minutes (12 hours): %s",
+        timezoneOffsetMinutes);
+    ZonedDateTime day = generateDay(instant, timezoneOffsetMinutes);
+    ZoneId zoneId = day.getZone();
+    String dayOfWeek = day.getDayOfWeek().getDisplayName(TextStyle.SHORT, Locale.US);
+    int year = day.getYear();
+    int month = day.getMonthValue();
+    int dayOfMonth = day.getDayOfMonth();
+
+    String formattedDate = formatDate(dayOfWeek, month, dayOfMonth);
+
+    ImmutableList.Builder<AvailabilityTimeSlot> timeSlots = ImmutableList.builder();
+    for (HoursAndMinutes hoursAndMinutes : ALL_HOURS_AND_MINUTES) {
+      int hour = hoursAndMinutes.hour();
+      int minute = hoursAndMinutes.minute();
+      LocalDateTime localTime = LocalDateTime.of(year, month, dayOfMonth, hour, minute);
+      ZonedDateTime utcTime =
+          ZonedDateTime.of(localTime, zoneId).withZoneSameInstant(ZoneOffset.UTC);
+      String formattedUTCTime = utcTime.format(DateTimeFormatter.ISO_INSTANT);
+
+      int standardHour = hour;
+      if (hour > 12) {
+        standardHour = hour - 12;
+      }
+      String formattedTime =
+          String.format("%d:%02d %s", standardHour, minute, hour < 12 ? "AM" : "PM");
+
+      timeSlots.add(
+          AvailabilityTimeSlot.create(
+              formattedUTCTime, formattedTime, formattedDate, getSelectedStatus(formattedUTCTime)));
+    }
+
+    return timeSlots.build();
+  }
+
+  // Uses an Instant and a timezoneOffsetMinutes int to create a ZonedDateTime instance.
+  private static ZonedDateTime generateDay(Instant instant, int timezoneOffsetMinutes) {
+    return instant.atZone(ZoneId.ofOffset("UTC", convertIntToOffset(timezoneOffsetMinutes)));
+  }
+
+  // This method takes the timezoneOffsetMinutes int and converts it
+  // into a proper ZoneOffset instance.
+  private static ZoneOffset convertIntToOffset(int timezoneOffsetMinutes) {
+    return ZoneOffset.ofHoursMinutes((timezoneOffsetMinutes / 60), (timezoneOffsetMinutes % 60));
+  }
+
+  // Returns a readable date string such as "Tue 7/7".
+  private static String formatDate(String dayOfWeek, int month, int dayOfMonth) {
+    return String.format("%s %d/%d", dayOfWeek, month, dayOfMonth);
+  }
+
+  // TODO: Access the time slot from data store using the utcEnconding to tell if it
+  // is selected or not.
+  // This method will tell whether or not a time slot has already been selected. (See
+  // TODO above).
+  private static boolean getSelectedStatus(String utcEncoding) {
```

Comment:
No change requested, but you won't be able to make this static in the real implementation because you'll need a DAO.

(@jalexanderqed  request to make all these methods results in non idiomatic code, as you'll find out in a subsequent PR)

Reply:
Ah, this method either wasn't here or I failed to notice it when I made the comment. This could still be resolved by passing in the DAO to the static method, but I think at that point it will be a style choice whether to make the methods non-static or pass in the DAO.